### DISCUSSION TITLE
Option to set delay for suggestions would be nice

### DISCUSSION THREAD

**Discussion by Mahasvan** [community/community#9978] [Copilot Conversations]
Copilot suggesting code immediately after I enter a character means I am unable to use the IntelliSense provided in PyCharm sometimes. 

[Link to video demonstrating what I mean](https://cdn.discordapp.com/attachments/848213319604633600/930450271979581541/demonstration.mp4)

The image below 👇 would be what I would have wanted to use.
![image](https://user-images.githubusercontent.com/82939599/148944750-679cee3e-b92e-4539-8088-3e0bc0e197a5.png)

It would be nice if Copilot had an a delay in suggesting code, or it just doesn't show suggestions when the intellisense panel is open

**Comment by fblanc92** [comment]
Absolutely. Pair programming is not about accepting a complete suggestion or not. 
It should have the option to delay them or just to be active but not showing suggestions in the same file.

**Comment by luketych** [comment]
I agree. If only the copilot team had basic ability to understand another person other than themselves lol

**Comment by cmp-nct** [comment]
I was surprised not to find any config settings for such a complex tool.
When adding a delay to the completion copilot should search immediately so it's not delayed beyond the configuration time.
For example: I'd like copilot to search for completions with automated completions being disabled.
So when I press/click for "completion" it comes immediately as it was already prepared.

**Comment by felipelalli** [comment]
I'd like to increase to 5~30 seconds, depending of the file.

**Comment by CiberNin** [comment]
The new update might help with this.
The put a lil AI clientside that should throttle suggestions when they don't seem usefull.

**Comment by nyteshade** [comment]
OMG Please please please allow for a user specified delay before suggestions come in. I find myself often disabling the help because it wont shut up, keeps getting it wrong or doesn't let me think about the direction I want to explore. Furthermore, I'd really like a button to ask for help rather than it just "autoplaying" the suggestion

**Comment by pnacelli** [comment]
Glad this suggestion already exists.. just checking if this is being looked at for all JetBrain products or just PyCharm as originally reported by OP?

**Comment by beatread** [comment]
I don't know whether it will be useful for someone, but if I put the comma right after cursor, then it prevents copilot from completions. Like this:
`myClassVar.doSmth*cursor is here, and right after a comma*,`

**Comment by FrancoisGrosbellet** [comment]
+1 here, a delay feature would really be appreciated. Honestly, the fact that feature does not exist is currently Copilot's biggest issue for my usage.

Very similar request here: [Add a suggestion delay option in the configuration #17670](https://github.com/orgs/community/discussions/17670)

**Comment by nicksneiderdev** [comment]
Yeah this is still a needed change. I love being able to use suggestions in markdown files but the suggestions are too repetitive and fast that they are actually distracting. The code suggestion timing is OK. Would be fantastic to be able to set the suggestion delay depending on file type.

**Comment by gjlmotea** [comment]
Wish there was this feature, otherwise I would disable the copilot sometimes to select autocompletion.

**Comment by bdmorin** [comment]
How is this not already implemented? immediate inline suggestions are unwanted. Gimme 30 seconds to look at my code!

**Comment by jacsamg** [comment]
I really need this, it's distracting me a lot during focus sessions.

**Comment by quangv** [comment]
Could be related: https://github.com/orgs/community/discussions/17670

Seems like a popular request.

**Comment by kyunussov** [comment]
please implement
